File: android/app/src/main/kotlin/io/github/evokelektrique/tunnelforge/ProfileTransferContract.kt
package io.github.evokelektrique.tunnelforge

/** Method channel contract for incoming `tf://` links and `.tfp` profile files. */
object ProfileTransferContract {
    const val METHOD_CHANNEL = "io.github.evokelektrique.tunnelforge/profileTransfer"

    const val CONSUME_PENDING_TRANSFERS = "consumePendingTransfers"
    const val ON_INCOMING_TRANSFER = "onIncomingTransfer"

    const val ARG_TYPE = "type"
    const val ARG_DATA = "data"
    const val ARG_MESSAGE = "message"
    const val ARG_SOURCE = "source"

    const val TYPE_TFP_JSON = "tfpJson"
    const val TYPE_TF_URI = "tfUri"
    const val TYPE_ERROR = "error"
}

File: lib/features/profiles/data/profile_transfer_contract.dart
/// Host/channel contract for Android profile transfer intents and file handoff.
abstract final class ProfileTransferContract {
  static const String channel =
      'io.github.evokelektrique.tunnelforge/profileTransfer';

  /// Flutter -> host: returns a list of pending transfer/error maps.
  static const String consumePendingTransfers = 'consumePendingTransfers';

  /// Host -> Flutter: pushes one transfer/error map after an external open/share event.
  static const String onIncomingTransfer = 'onIncomingTransfer';

  static const String argType = 'type';
  static const String argData = 'data';
  static const String argMessage = 'message';
  static const String argSource = 'source';

  static const String typeTfpJson = 'tfpJson';
  static const String typeTfUri = 'tfUri';
  static const String typeError = 'error';
}
